src/lib/pubkey/rsa/rsa.cpp

```diff
@@ -445,6 +445,11 @@ class RSA_Private_Operation {
             m_max_d1_bits(m_private->p_bits() + m_blinding_bits),
             m_max_d2_bits(m_private->q_bits() + m_blinding_bits) {}
 
+      std::span<uint8_t> raw_op(std::span<const uint8_t> input) {
+         auto sv = raw_op(input.data(), input.size());
+         return std::span<uint8_t>(sv.begin(), sv.end());
```

Comment:
This doesn't seem right - `sv` is a temporary within this function so on return the `span` will point to deallocated memory.

FWIW I am not sure exactly what the comment is suggesting to do - @reneme do you remember?

Reply:
For the record: I agree that this isn't correct (it would be good if tooling would warn about it).

> I am not sure exactly what the comment is suggesting to do

I'm assuming that you mean

```C++
// TODO: avoid copy by letting raw_op() work on std::span<>
```

which hints at making `raw_op()` work on a `std::span<uint8_t>` out-param just like `raw_kem_decrypt()` does. This would save a copy.

Meanwhile @devnexen adapted the patch accordingly.